### DISCUSSION TITLE
集群版部署3个proxy，注入单个proxy pod的cpu满压，会导致所有的proxy的延迟都明显升高。可以分析一下原理么。

### DISCUSSION THREAD

**Discussion by qiqiqi998** [milvus-io/milvus#38732] [Q&A and General discussion]
11点32分，对其中一个proxy的pod注入CPU压力，所有proxy的延迟明显升高，外部注入的POD CPU满压，而不是真实使用。
11点42左右，闭了写操作，延迟下降了一点。
下图是时延变化情况和一些查询相关的监控。
![图片](https://github.com/user-attachments/assets/a7b5a8cf-8551-493f-a111-1556bcdd0ec0)
![图片](https://github.com/user-attachments/assets/7b59e0c3-b5a1-4054-9b93-2577ec4950b4)
![图片](https://github.com/user-attachments/assets/486db802-17ed-4619-be2d-d71bf5fc709a)
![图片](https://github.com/user-attachments/assets/c5d6835d-5dfb-4714-884c-1fb1e6704b39)
![图片](https://github.com/user-attachments/assets/0c6a71ad-0f3f-4d77-b6ee-3b32fdf20192)

下图是POD情况。
![图片](https://github.com/user-attachments/assets/b6ce5eef-e0c8-456b-935b-0a72941c4691)

日志信息。
[logs.tar.gz](https://github.com/user-attachments/files/18243128/logs.tar.gz)

**Comment by yhmo** [comment]
注入CPU压力是什么意思？看起来查询请求数量和insert请求数量没什么变化

**Reply by qiqiqi998** [reply]
通过外部工具，将pod的CPU打满，制造类似k8s节点CPU打满问题。这里只将一个proxy的CPU打满了，会导致所有proxy的延迟都明显升高。
![图片](https://github.com/user-attachments/assets/4d9c5973-570b-4899-b349-bd727f118323)

关闭写之前，延迟都很高，关闭写之后，在没注入故障的proxy上会出现持续的上下浮动。
![图片](https://github.com/user-attachments/assets/c86fe33a-6fd9-4a53-8a18-00ebc7f8e8dd)

**Reply by qiqiqi998** [reply]
proxy之间应该是独立的吧，为什么单个proxy会导致整个系统的查询延迟都升高，查询一致性等级用的是eventually。

**Comment by zhuwenxing** [comment]
你看看其他两个proxy是否有流量请求？milvus 多proxy你需要在proxy 的service加上一个负载均衡
目前pymivlus的连接是单连接，所以请求可能都会发送到同一个proxy上。

**Reply by qiqiqi998** [reply]
测试脚本的查询是多线程，一个线程一个链接。写入是单线程写入，每次写入几条。
可以看监控面板的截图，所有proxy上都是有查询的，并且查询延迟升高也是一起出现的。
![图片](https://github.com/user-attachments/assets/c9f43de6-4db6-4fbd-92bd-d7e96ec8293c)

**Reply by zhuwenxing** [reply]
好的，我们会尝试本地复现一下
